lib/LanguagePicker.js

```diff
@@ -76,43 +76,48 @@ LanguagePicker.prototype.newProcessor = function newProcessor() {
       }
 
       // 2. Fallback language from fallbacks.json
-      for (let i = 0; i < this.fallbacks.length; i++) {
-        if (this.values[this.fallbacks[i]]) {
-          return this.values[this.fallbacks[i]];
+      for (const fallback of this.fallbacks) {
+        val = this.values[fallback];
+        if (val) {
+          return val;
         }
       }
 
       // 3. Any language with suffix of same script as requested language
-      for (let i = 0; i < valueLangCodes.length; i++) {
-        if (valueLangCodes[i].endsWith(this.langScript)) {
-          return this.values[valueLangCodes[i]];
+      const valueLangCodes = Object.keys(this.values);
+      for (const langCode of valueLangCodes) {
+        val = this.values[langCode];
+        if (val && langCode.endsWith(this.langScript)) {
+          return val;
         }
       }
 
       // 4. Another language with the same script
-      for (let i = 0; i < this.languagesInScript.length; i++) {
-        if (this.values[this.languagesInScript[i]]) {
-          return this.values[this.languagesInScript[i]];
+      for (const langCode of this.languagesInScript) {
+        val = this.values[langCode];
+        if (val) {
+          return val;
         }
       }
 
       // 5. English
-      if (this.values[this.prefix ? `${this.prefix}en` : 'en']) {
-        return this.values[this.prefix ? `${this.prefix}en` : 'en'];
+      val = this.values[this.prefix ? `${this.prefix}en` : 'en'];
```

Comment:
i think it would also be good to make `this.prefix ? `${this.prefix}en` : 'en'` into a class-level constant -- no point in constructing a string for every tag - string concatenations are expensive when doing hundreds of millions of them.